### PR TITLE
Cart: Final Merge

### LINKED ISSUE #192: Update user "cart" to allow a greater number samples
Based on the way the current cart system functions, when users add a large number of samples to the cart the interface becomes sluggish.

**TODOs**

* [x]  UI:
  * [x] Remove client side code for slide out cart
  * [x] Cart Page
* [x] Server:
  * [x] Add samples to cart
  * [x] Remove samples from cart
  * [x] Remove project from cart
* [x]  Fix Tests:
  * [x] CartControllerTest
  * [x] PipelinesPhylogenomicsPageIT
  * [x] AssemblyPipelinePageIT
  * [x] CartControllerIT

### PR DESCRIPTION
**Pull request by joshsadam**
## Description of changes
Complete update of the IRIDA cart system.

## Related issue
Closes #192 

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

* [x] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.
* [x] Tests added (or description of how to test) for any new features.
* [x] User documentation updated for UI or technical changes.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,6 +24,7 @@ Changes
 * [UI/Developer]: Fixed issue where search for member to add to group resulted in no results found when search term contained capital letters.
 * [Documentation]: Added information on fixing ONLY_FULL_GROUP_BY sql error to the administrator faq docs.
 * [UI]: User on a remote project with a project role of manager has the ability to assign user groups to the project.
+* [UI]: New dedicated cart page.
```
Comment (apetkau) [inline CHANGELOG.md]: Could you add a statement also saying that we've updated the interface for exporting to Galaxy, which now must be done through the cart?

File: doc/_includes/tutorials/common/samples/galaxy-export.md [hunk 2]
```diff
@@ -12,17 +12,15 @@ After you log in to IRIDA (or if you were already logged in), you will be direct
 
 ![Galaxy IRIDA projects list.]({{ site.baseurl }}/images/tutorials/common/samples/galaxy-irida-projects-list.png)
 
-Navigate to the list of samples that you're interested in exporting by clicking on the project name. Then, [select the samples]({{ site.baseurl }}/user/user/samples/#selecting-samples) that you want to export, click on the "Export" button just above the samples list, then click "Send to Galaxy":
```
Comment (apetkau) [inline doc/_includes/tutorials/common/samples/galaxy-export.md]: I noticed that the **Galaxy data library** link is broken at the top of this page. I realize it's not something that you touched, but could you update it to point to <https://galaxyproject.org/data-libraries/>?

File: doc/user/user/pipelines/index.md [hunk 3]
```diff
@@ -18,48 +18,41 @@ Adding samples to the cart
...
-![Full cart.](images/full-cart.png)
+![Pipeline Cart.](images/pipeline-launch-cart.png)
 
-You can remove individual samples or all selected samples from a project by clicking on the <img src="images/remove-icon.png" class="inline" alt="Remove icon."> icon, or you can empty the cart entirely by clicking on the "Clear" button, in the top, right-hand corner of the cart:
+You can remove individual samples or all selected samples from a project by clicking on the sample menu icon:
 
-![Cart clear button.](images/cart-clear-button.png)
+ ![Cart clear button.](images/menu-icon.png)
+ 
+And either selection “Remove Sample” which will remove just a single sample or “Remove Project” which will remove all samples that belong to the same project. You can also empty the cart entirely by clicking on the “Empty” button, in the top, right-hand corner of the cart.
```
Comment (apetkau) [inline doc/user/user/pipelines/index.md]: either **selection** -> either **selecting**

File: CHANGELOG.md (hunk 1 above)
Comment (joshsadam) [reply CHANGELOG.md]: Updated in c96767bc71fb37d639ea0713566d5aefb5fcaada

File: doc/_includes/tutorials/common/samples/galaxy-export.md (hunk 2 above)
Comment (joshsadam) [reply doc/_includes/tutorials/common/samples/galaxy-export.md]: Updated in 00f7f59d6e7217541ac0b79043f776d22b282d38

File: doc/user/user/pipelines/index.md (hunk 3 above)
Comment (joshsadam) [reply doc/user/user/pipelines/index.md]: Updated in 88ad188be78219b0dca0730d47971f171daa3a18

File: CHANGELOG.md (hunk 1 above)
Comment (apetkau) [reply CHANGELOG.md]: Thanks. Could you also say in that statement that we've deprecated exporting to Galaxy through the Project Samples page?

Comment (joshsadam) [reply CHANGELOG.md]: Updated in 55ebdd8315e756894c5edd1fd6a8e5d34ea84099

**Comment by joshsadam** [comment]
@apetkau I think I hit everything you commented on.  The pipelines page I am still considering, and might do that is a new, of development, branch.  Should be pretty easy once this is in.  I did update the ajax url so that the user would not just get a json printout if they had it bookmarked.

**Comment by apetkau** [comment]
Thanks @joshsadam. Yes, the `/pipelines` page is good now. And the updated screenshots in the documentation looks great.

Could you update these sections in our documentation?

* Cart screenshot in https://github.com/phac-nml/irida/blob/cart/_base/doc/user/user/dashboard/index.md#cart
* select-a-pipeline.png screenshot in https://github.com/phac-nml/irida/blob/cart/_base/doc/_includes/tutorials/common/selecting-pipeline.md
    * It looks like it got deleted, but it's still being referenced. I think you can delete the text referencing this image.
* Add sample to cart screenshto in https://github.com/phac-nml/irida/blob/cart/_base/doc/user/tutorials/refseq_masher/index.md#add-sample-to-cart
* Select a pipeline screenshot and text in https://github.com/phac-nml/irida/blob/cart/_base/doc/user/tutorials/mentalist/index.md#selecting-a-pipeline

Also, could you merge `development` into this branch and fix the `CHANGELOG.md` merge conflict?

**Comment by joshsadam** [comment]
Thanks @apetkau I just finished those last images and documentation for the tutorials.

**Comment by apetkau** [comment]
All the documentation looks good. Thanks @joshsadam 

One more comment I have. If I add samples to the cart, then click **Empty Cart**, I get the page showing all pipelines and an empty cart, but the **Select** button is still displayed on each pipeline:

image

I believe these should be hidden in this situation, correct?

**Comment by joshsadam** [comment]
Fixed the issue with the select buttons in 7dfba4f87f0ca43245e615da679ba8644ffc0991, thanks

**Comment by apetkau** [comment]
Thanks @joshsadam. Everything looks good to me. I will merge whenever the tests past.

**Comment by apetkau** [comment]
This looks great @joshsadam. Thank you for all the hard work you (and others) put into getting the tests to work.

**Pull request merged**